### ISSUE TITLE
Job does not run at fixed interval

### ISSUE THREAD

**Issue by satishmane** [vikeri/react-native-background-job#19]
Hi vikeri,

I want to run a JS task at interval of every 2 minute or so while app is in background or when rebooted. Job runs on reboot or in background always and schedules tasks. This works fine. Only issue I am facing is that it does not start JS task every two minutes. It will start JS task at 2min, 5min or 7min or max 18 mins or so. 

Is there way to ensure JS task runs always consistently at fix interval overwriting previous task?
Thanks
Satish

**Comment by vikeri** [comment]
What version of Android are you running? As stated here: [`schedule`](https://github.com/vikeri/react-native-background-job/tree/554787ec25a31b596e6b2f9e941f786c056e8f74#schedule) if you run Android N or later it will not respect the value you set but instead set it to approximately 15 minutes.

I'm currently working on an, "always on" mode though that permits very exact intervals, it's still not production ready but a beta release is available [here](https://github.com/vikeri/react-native-background-job/releases/tag/v1.1.3-0)

**Comment by satishmane** [comment]
Thank you very much for quick reply.
Current I am pre-Android N device which has Android 5.1 version. 
I also downloaded beta version today. I did not basic testing of beta version and it works fine. I need to do testing leaving job running for 24 hours and see whether it starts JS tasks at exact interval.

**Comment by vikeri** [comment]
Any feedback on this? Can it be closed?

**Comment by satishmane** [comment]
Hi,
Yes, have been testing for last 4-5 days. 

With beta version, job starts at fixed interval for few hours and but then after few hours, it starts bit early. I had configured it to run every 2 mins and left it running overnight. It used to start every two minutes for 8 hours throughout night. But in the morning, when I used mobile for some hours and then checked log, I found job running every minute. 

Earlier with master branch version, job used to get delayed i.e. if configured for 2 minutes, then it used to take more time than 2 min to start next job. Sometime it will run at 2 minute and sometime it will take 5, 10 etc.

In short, like cron job, it does not start job task exactly at configured interval.

thanks

**Comment by angoel** [comment]
Hi,

I have been running this application for past 48 hours where this job has been getting fired at every 2 minutes. After 48 hours it has been noticed that phone has become slow and response in opening of the application has increased (e.g. to open gmail or chrome it takes a while to come up).

Thanks
Anil

**Comment by vikeri** [comment]
@angoel It might be that there's a memory leak in your JS code that could cause this. Have you checked that?

**Comment by vikeri** [comment]
No activity for a long time. Closing. Please create new issue if re-encountered.
